packages/blade/src/components/Collapsible/CollapsibleButton.tsx

```diff
@@ -0,0 +1,53 @@
+import type { ReactElement } from 'react';
+import { useCallback } from 'react';
+import { useCollapsibleContext } from './CollapsibleContext';
+import type { ButtonProps } from '~components/Button';
+import { Button } from '~components/Button';
+import type { IconComponent } from '~components/Icons';
+
+type CollapsibleButtonProps = Pick<
+  ButtonProps,
+  | 'variant'
+  | 'size'
+  | 'iconPosition'
+  | 'isDisabled'
+  | 'testID'
+  | 'accessibilityLabel'
+  | 'icon'
+  | 'children'
+>;
+
+const CollapsibleButton = ({
+  children,
+  variant,
+  size,
+  icon,
+  iconPosition,
+  isDisabled,
+  testID,
+  accessibilityLabel,
+}: CollapsibleButtonProps): ReactElement => {
+  const { setIsExpanded } = useCollapsibleContext();
+
```

Comment:
add 
```
TODO: throw error if CollapsibleButton is used without Collapsible context and same for other sub components
```
for your follow up PRs

Reply:
might be a miss ideally we should, its better to give feedback rather than leaving users clueless that why isn't it working. also some less "slack pings" 😆

Reply:
Done, have added `null` context checks for both `useCollapsible` and `useAccordion` so using any of these subcomponents outside should now throw error.

Reply:
Done in #1296 (the context checks)